TACC/Core-Portal
# Task/WP-725: Mutation Hooks: Extract Files

## Overview

We are working on updating our current mutations to utilize TypeScript and React Query instead of React Redux. This should improve the speed and functionality of all portal systems going forward.

## Related

* [WP-725](https://tacc-main.atlassian.net/browse/WP-725)

## Changes

I converted `useExtract.js` to `useExtract.ts` and created a mutation that extracts files from a ZIP file into the directory where the ZIP file is currently located.

## Testing

1. Navigate to a ZIP file in any directory.
2. Select that ZIP file and click on the Extract button.
3. In the modal that appears, click on the purple Extract button.
4. A job gets submitted to extract files from the ZIP file, and you'll eventually see the folder containing those files in that directory.

## UI

![image](https://github.com/user-attachments/assets/dde3eb2a-fd25-4112-bb08-05b3b1f78777)

## Notes
